hardware/raspberrypi/bootmodes/gpio.md

```diff
@@ -0,0 +1,33 @@
+# GPIO boot mode
+
+The Raspberry Pi can be configured to allow the boot mode to be selected at power on using hardware attached to the GPIO connector: this is GPIO boot mode. This is done by setting bits in the OTP memory of the SoC. Once the bits are set, they permanently allocate 5 GPIOs to allow this selection to be made. Once the OTP bits are set they cannot be unset so you should think carefully about enabling this, since those 5 GPIO lines will then be unavailable for any other use.
```

Comment:
The GPIO boot modes do not permanently allocate the GPIOs, you can still use the pins afterwards for input or output.  To enable a mode it just requires a pull up to 3v3 strong enough to override the internal pull down (we suggest 5k).